metals/src/main/scala/scala/meta/internal/metals/DocumentHighlightProvider.scala

```diff
@@ -0,0 +1,162 @@
+package scala.meta.internal.metals
+
+import org.eclipse.lsp4j.DocumentHighlight
+import org.eclipse.lsp4j.DocumentHighlightKind
+import org.eclipse.lsp4j.TextDocumentPositionParams
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.mtags.DefinitionAlternatives.GlobalSymbol
+import scala.meta.internal.mtags.Semanticdbs
+import scala.meta.internal.mtags.Symbol
+import scala.meta.internal.semanticdb.Scala.Descriptor
+import scala.meta.internal.semanticdb.Scala.Symbols
+import scala.meta.internal.semanticdb.SymbolInformation
+import scala.meta.internal.semanticdb.SymbolOccurrence
+import scala.meta.internal.semanticdb.TextDocument
+
+final class DocumentHighlightProvider(
+    definitionProvider: DefinitionProvider,
+    semanticdbs: Semanticdbs
+) {
+
+  def documentHighlight(
+      params: TextDocumentPositionParams
+  ): java.util.List[DocumentHighlight] = {
+    val source = params.getTextDocument.getUri.toAbsolutePath
+    val result = semanticdbs.textDocument(source)
+
+    val highlights = for {
+      doc <- result.documentIncludingStale.toList
+      positionOccurrence = definitionProvider.positionOccurrence(
+        source,
+        params,
+        doc,
+        includeLastCharacter = true
+      )
+      occ <- positionOccurrence.occurrence.toList
+      alternatives = findAllAlternatives(doc, occ)
+      curr <- doc.occurrences
+      if curr.symbol == occ.symbol || alternatives(curr.symbol)
+      range <- curr.range
+      revised <- positionOccurrence.distance.toRevised(range.toLSP)
+      kind = if (curr.role.isDefinition) DocumentHighlightKind.Write
```

Comment:
nit: wrap this if/else in braces to workaround the weird indentation, I never got around to fix this in scalafmt

Reply:
added the braces